addons/mail/static/src/components/thread_view/thread_view.xml

```diff
@@ -22,6 +22,16 @@
                         t-ref="messageList"
                     />
                 </t>
+                <t t-elif="threadView.threadCache.hasLoadingFailed">
+                    <div class="o_ThreadView_loadingFailed">
+                        <div class="o_ThreadView_loadingFailedText">
+                            An error occured while fetching messages.
+                        </div>
+                        <button class="btn btn-link o_ThreadView_loadingFailedBtn" t-on-click="_onClickRetryLoadMessages">
```

Comment:
Structural classnames (`o_ComponentName_partName`) before styling classnames (`btn`, `btn-link`).
Also I prefer full writing of "Button" rather than short-form "Btn" in part name.